sdk/eventhub/azure-eventhubs/azure/eventhub/eventprocessor/event_processor.py

```diff
@@ -0,0 +1,170 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# -----------------------------------------------------------------------------------
+
+from typing import Callable, List
+import uuid
+import asyncio
+import logging
+
+from azure.eventhub import EventPosition, EventHubError
+from azure.eventhub.aio import EventHubClient
+from .checkpoint_manager import CheckpointManager
+from .partition_manager import PartitionManager
+from .partition_processor import PartitionProcessor, CloseReason
+from .utils import get_running_loop
+
+log = logging.getLogger(__name__)
+
+
+class EventProcessor(object):
+    def __init__(self, eventhub_client: EventHubClient, consumer_group_name: str,
+                 partition_processor_factory: Callable[..., PartitionProcessor],
+                 partition_manager: PartitionManager, **kwargs):
+        """An EventProcessor automatically creates and runs consumers for all partitions of the eventhub.
+
+        It provides the user a convenient way to receive events from multiple partitions and save checkpoints.
+        If multiple EventProcessors are running for an event hub, they will automatically balance loading. This feature
+        won't be availabe until preview 3.
+
+        :param consumer_group_name: the consumer group that is used to receive events
+        from the event hub that the eventhub_client is going to receive events from
+        :param eventhub_client: an instance of azure.eventhub.aio.EventClient object
+        :param partition_processor_callable: a callable that is called to return a PartitionProcessor
+        :param partition_manager: an instance of a PartitionManager implementation
+        :param initial_event_position: the offset to start a partition consumer if the partition has no checkpoint yet
+        """
+        self._consumer_group_name = consumer_group_name
+        self._eventhub_client = eventhub_client
+        self._eventhub_name = eventhub_client.eh_name
+        self._partition_processor_factory = partition_processor_factory
+        self._partition_manager = partition_manager
+        self._initial_event_position = kwargs.get("initial_event_position", "-1")
+        self._max_batch_size = eventhub_client.config.max_batch_size
+        self._receive_timeout = eventhub_client.config.receive_timeout
+        self._tasks: List[asyncio.Task] = []
+        self._instance_id = str(uuid.uuid4())
+        self._partition_ids = None
```

Comment:
What is `self._partition_ids`? It's "private" and not used from what I can tell?

Reply:
I declared self._partitions_ids but afterwards I set value to self.partition_ids and read it by mistake. And it happens to work with Python. Corrected to use self._partition_ids.